ci/setup_version.py

```diff
@@ -0,0 +1,34 @@
+"""Add pre-release version to the package."""
+
+from packaging.version import parse
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Set the version of the package.")
+    parser.add_argument("version", type=str, help="The version to set.")
+    args = parser.parse_args()
+
+    with open("python/Cargo.toml", "r") as f:
+        lines = f.readlines()
+        for i, line in enumerate(lines):
+            if line.startswith("version = "):
+                current_version = line.split('"')[1]
+                lines[i] = f'version = "{args.version}"\n'
+                break
+        else:
+            raise ValueError("Could not find version in Cargo.toml")
+
+    parsed_version = parse(args.version)
+    current_version = parse(current_version)
+
+    assert (
+        parsed_version.release == current_version.release
+    ), f"Base versions do not match: {parsed_version.release} != {current_version.release}"
```

Comment:
Wait...why do these match?  Are we setting the version or asserting the version is correct?  Or is it because `current_version` and `parsed_version` are just the "non-beta" part of the version?

Reply:
The `.release` property is the non-beta part of the version. So we are just checking we aren't modifying the base version.